### DISCUSSION TITLE
Use one Event with two Projectors or two Events with one Projector.

### DISCUSSION THREAD

**Discussion by mascam97** [spatie/laravel-event-sourcing#317] [Q&A] [answered]
I have an electronic wallet, where each user has a pocket with money and can transfer money to another user.

Case: When a user transfer money the system creates a transaction, and on another hand when a user receives the money there is also another transaction. So at the final, the entire process creates two transactions (one per user) with almost the same data. Both transactions are stored with a groupUuid in order to know the correlation and causation transactions.

Problem: How could I make the Event according to event sourcing?

Possible solutions:

1. **Create one event**; I stored two transactions UUID (incomeTransactionUuid and egressTransactionUuid) with both Pockets (egressPocketUuid and incomePocketUuid) and all the data that belongs to the main Action. And then create two projectors (IncomeTransaction and egressTransaction) that use the same event and read in some cases different data.
2. **Create two events**; each event for each transaction and its own UUIDs (transactionUuid and pocketUuid), with one projector (TransactionProjector). This solution repeats the same data like the amount, concept, and currency in both events, and the projector would read all data in the event.

The first solution is better for me, according to the book, an event is a source of truth and it has all the entire process. Both options would work well, it's just a logic problem. Which option looks better? What do you think?

Thanks :)

**Comment by brendt** [comment, marked as answer]
Always choose what's closest to the real-world problem. In this case **one transaction** is creation, **from** someone, **to** someone. So it's one event with two (and more) fields.

**Reply by mascam97** [reply]
In this case **two transactions** are created, an income transaction and an egress transaction. The transactions have an attribute called **movement** (income or egress), each transaction belongs to one user (from and to someone), and both have a groupUuid to relate them to each other.

Choosing what´s closest to the real-world problem, would be two events

Thanks :)

**Reply by brendt** [reply]
Awesome! Than yes, I agree
